core/src/main/scala/scala/collection/parallel/ParSeqLike.scala

```diff
@@ -62,8 +62,8 @@ extends ParIterableLike[T, CC, Repr, Sequential]
     *            this sequence in the same order, `false` otherwise
     */
   override def equals(that: Any): Boolean = that match {
-    case that: ParSeq[_] => (that eq this.asInstanceOf[AnyRef]) || (that canEqual this) && (this sameElements that)
-    case _               => false
+    case that: ParSeq[Any] => (that eq this.asInstanceOf[AnyRef]) || (that canEqual this) && (this sameElements that)
```

Comment:
Excellent interview question on what type in this position does not warn unchecked.

It took me a minute to see that `that` is shadowed, which is why that idiom annoys me endlessly.

Why do people complain about "my editor needs braces, I can't go braceless" but not "my editor doesn't show shadowing, I can't use shadows".